src/dlls/mscoree/mscoree.cpp

```diff
@@ -907,9 +907,23 @@ STDAPI GetCORSystemDirectoryInternal(__out_ecount_part_opt(cchBuffer, *pdwLength
     if (pBuffer == NULL)
         IfFailGo(E_POINTER);
 
+#ifdef CROSSGEN_COMPILE
```

Comment:
Should we initialize the PAL using PAL_InitializeCoreCLR then?

PAL_InitializeCoreCLR calls InitializeStringResources - that should be done for crossgen too to get good error messages.